src/components/KInput/KInput.vue

```diff
@@ -91,81 +91,39 @@
 
 <script setup lang="ts">
 import { computed, ref, watch, useSlots, useAttrs, onMounted, nextTick, useId } from 'vue'
-import type { PropType } from 'vue'
-import type { LabelAttributes, LimitExceededData } from '@/types'
+import type { InputProps, InputEmits, InputSlots } from '@/types'
 import useUtilities from '@/composables/useUtilities'
 import KLabel from '@/components/KLabel/KLabel.vue'
 import { KUI_ICON_SIZE_40 } from '@kong/design-tokens'
 import { VisibilityIcon, VisibilityOffIcon } from '@kong/icons'
 
-const props = defineProps({
-  modelValue: {
-    type: [String, Number],
-    default: '',
-  },
-  label: {
-    type: String,
-    default: '',
-  },
-  labelAttributes: {
-    type: Object as PropType<LabelAttributes>,
-    default: () => ({}),
-    validator: (value: LabelAttributes): boolean => {
-      if (value.help) {
-        console.warn('KInput: `help` property of `labelAttributes` prop is deprecated. Please use `info` prop instead. See the migration guide for more details: https://kongponents.konghq.com/guide/migrating-to-version-9.html#klabel')
-      }
-
-      return true
-    },
-  },
-  help: {
-    type: String,
-    default: '',
-  },
-  error: {
-    type: Boolean,
-    default: false,
-  },
-  errorMessage: {
-    type: String,
-    default: '',
-  },
-  characterLimit: {
-    type: Number,
-    default: null,
-    // Ensure the characterLimit is greater than zero
-    validator: (limit: number): boolean => limit > 0,
-  },
-  /**
-   * @deprecated in favor of `error`
-   */
-  hasError: {
-    type: Boolean,
-    default: false,
-    validator: (value: boolean): boolean => {
-      if (value) {
-        console.warn('KInput: `hasError` prop is deprecated. Please use `error` prop instead. See the migration guide for more details: https://kongponents.konghq.com/guide/migrating-to-version-9.html#kinput')
-      }
+const {
+  modelValue = '',
+  label = '',
+  labelAttributes = {},
+  help = '',
+  error = false,
+  errorMessage = '',
+  characterLimit = null,
+  hasError = false,
+  type = 'text',
+  showPasswordMaskToggle = false,
```

Comment:
to be like the other PRs, we shouldn't need to define the `false` prop defaults here for boolean props, right?